vnext/Folly/minfutureimpl/folly/futures/Promise.h

```diff
@@ -0,0 +1,461 @@
+/*
```

Comment:
Would you be able to point out the changes that were made compared to the original?

The full size of the minimal future implementation still seems quite large. I worry a bit about how we can keep this up to date (e.g. if the folly headers this relies on changes their API).

Reply:
I'm concerned about the size and upkeep too; I would love to see the changes upstream before we consume them here. It would be hard to guarantee that our custom stuff doesn't get broken as changes upstream flow in, at least if we are able to upstream it we can guarantee that breaks are caught in the folly repo the minute they happen

Reply:
Thanks for sharing these vital concerns .. Yes, the current "min" implementation is not really minimal.. and it's risky for it have dependency on folly headers .. Let me try to make the minfuture implementation standalone and truly minimal.

Reply:
We no longer have the minimal implementation .. Even in the hermes build scripts that we use, we don't have any custom implementation, instead we use the old folly code .. All the old folly code is contained within the dll public api surface ..